internal/pkg/cli/deploy/workload.go

```diff
@@ -253,30 +262,32 @@ func newWorkloadDeployer(in *WorkloadDeployerInput) (*workloadDeployer, error) {
 
 	cfn := cloudformation.New(envSession, cloudformation.WithProgressTracker(os.Stderr))
 	return &workloadDeployer{
-		name:               in.Name,
-		app:                in.App,
-		env:                in.Env,
-		image:              in.Image,
-		resources:          resources,
-		workspacePath:      ws.Path(),
-		fs:                 &afero.Afero{Fs: afero.NewOsFs()},
-		s3Client:           s3.New(envSession),
-		addons:             addons,
-		imageBuilderPusher: imageBuilderPusher,
-		deployer:           cfn,
-		tmplGetter:         cfn,
-		endpointGetter:     envDescriber,
-		spinner:            termprogress.NewSpinner(log.DiagnosticWriter),
-		templateFS:         template.New(),
-		envVersionGetter:   in.EnvVersionGetter,
-		overrider:          in.Overrider,
-		customResources:    in.customResources,
-
+		name:                     in.Name,
+		app:                      in.App,
+		env:                      in.Env,
+		image:                    in.Image,
+		resources:                resources,
+		workspacePath:            ws.Path(),
+		fs:                       &afero.Afero{Fs: afero.NewOsFs()},
+		s3Client:                 s3.New(envSession),
+		addons:                   addons,
+		repository:               repository,
+		deployer:                 cfn,
+		tmplGetter:               cfn,
+		endpointGetter:           envDescriber,
+		spinner:                  termprogress.NewSpinner(log.DiagnosticWriter),
+		templateFS:               template.New(),
+		envVersionGetter:         in.EnvVersionGetter,
+		overrider:                in.Overrider,
+		customResources:          in.customResources,
 		defaultSess:              defaultSession,
 		defaultSessWithEnvRegion: defaultSessEnvRegion,
 		envSess:                  envSession,
 		store:                    store,
 		envConfig:                envConfig,
+		dockerCmdClient:          dockerCmdClient,
```

Comment:
what do you think about embedding this in `repository` instead? it seems like it's only used when calling functions on `repository` anyways

Reply:
Done. moved to `repository` pkg.